sdk/iot/hub/samples/telemetry/src/iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,278 @@
+      "IoT Hub FQDN", IOT_HUB_FQDN, "", false, iot_hub_fqdn_span, &trusted));
+
+  AZ_RETURN_IF_FAILED(
+      create_mqtt_endpoint(mqtt_endpoint, (int32_t)sizeof(mqtt_endpoint), iot_hub_fqdn_span));
+
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_fqdn_span, 0, (int32_t)strlen(iot_hub_fqdn)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      NULL));
+
+  return AZ_OK;
+}
+
+static int connect_device()
+{
+  int rc;
+
+  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
+  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+
+  char username[128];
+  if ((rc = az_iot_hub_client_get_user_name(&client, username, sizeof(username), NULL)) != AZ_OK)
+
+  {
+    printf("Failed to get MQTT clientId, return code %d\n", rc);
+    return rc;
+  }
+
+  mqtt_connect_options.username = username;
+  mqtt_connect_options.password = NULL; // Using X509 Client Certificate authentication.
+
+  mqtt_ssl_options.keyStore = (char*)x509_cert_pem_file;
+  if (*x509_trust_pem_file != '\0')
+  {
+    mqtt_ssl_options.trustStore = (char*)x509_trust_pem_file;
+  }
+
+  mqtt_connect_options.ssl = &mqtt_ssl_options;
+
+  if ((rc = MQTTClient_connect(mqtt_client, &mqtt_connect_options)) != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to connect, return code %d\n", rc);
+    return rc;
+  }
+
+  return 0;
+}
+
+static int send_telemetry_messages()
+{
+  int rc;
+  for (int i = 0; i < NUMBER_OF_MESSAGES; ++i)
+  {
+    printf("Sending Message %i\n", i + 1);
+    if ((rc = MQTTClient_publish(
+             mqtt_client,
+             telemetry_topic,
+             (int)strlen(telemetry_message_payloads[i]),
+             telemetry_message_payloads[i],
+             0,
+             0,
+             NULL))
+        != MQTTCLIENT_SUCCESS)
+    {
+      printf("Failed to publish telemetry topic, return code %d\n", rc);
+      return rc;
+    }
+    sleep_seconds(1);
+  }
+  return rc;
+}
+
+int main()
+{
+  int rc;
+
+  if ((rc = read_configuration_and_init_client()) != AZ_OK)
+  {
+    printf("Failed to read configuration from environment variables, return code %d\n", rc);
+    return rc;
+  }
+
+  char client_id[128];
+  size_t client_id_length;
+  if ((rc
+       = az_iot_hub_client_get_client_id(&client, client_id, sizeof(client_id), &client_id_length))
+      != AZ_OK)
+  {
+    printf("Failed to get MQTT clientId, return code %d\n", rc);
+    return rc;
+  }
+
+  if ((rc = MQTTClient_create(
+           &mqtt_client, mqtt_endpoint, client_id, MQTTCLIENT_PERSISTENCE_NONE, NULL))
+      != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to create MQTT client, return code %d\n", rc);
+    return rc;
+  }
+
+  if ((rc = connect_device()) != 0)
+  {
+    return rc;
+  }
+
+  if ((rc = az_iot_hub_client_telemetry_get_publish_topic(
+           &client, NULL, telemetry_topic, sizeof(telemetry_topic), NULL))
+      != AZ_OK)
+  {
+    return rc;
+  }
```

Comment:
can we move this inside the send telemetry function, because looks like that uses this

Reply:
moved